### PR TITLE
Bug fix/internal issue #524

### PR DESCRIPTION
**Pull request by Dronplane**
Added tests for index creation link with inBackground parameter set.

https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/5145/

### CODE REVIEW

File: tests/js/common/aql/aql-view-arangosearch-ddl-rocksdb.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+              params: params
+            });
+          };
+      tasks.register({
+        command: commandText,
+        params: { colName, inTransCount, dbName },
+        name: "CreateLinkInBackgroundMode_AuxTask"
+      });
+
+      require('internal').sleep(5); // give transaction some time to run before us
```
Comment (jsteemann) [inline tests/js/common/aql/aql-view-arangosearch-ddl-rocksdb.js]: The 5 seconds here will not guarantee that the task with the transaction has already run or is completed when the sleep is over. If you want to ensure this (and the test depends on it) it should be solved in a different way.

File: tests/js/common/aql/aql-view-arangosearch-ddl-rocksdb.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  const db = require("@arangodb").db;
+  const dbName = 'TestDB';
+  return {
+    setUp: function() {
+      db._useDatabase("_system");
+      try { db._dropDatabase(dbName); } catch (e) {}
+      db._createDatabase(dbName);
+    },
+    tearDown: function() {
+      db._useDatabase("_system");
```
Comment (jsteemann) [inline tests/js/common/aql/aql-view-arangosearch-ddl-rocksdb.js]: the teardown should unregister the task `CreateLinkInBackgroundMode_AuxTask` (will throw if the task isn't there anymore, so it needs to be wrapped in a try...catch).

Comment (Dronplane) [reply tests/js/common/aql/aql-view-arangosearch-ddl-rocksdb.js]: Added unregistering.

**Comment by Dronplane** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/5151/

**Pull request merged**